src/components/include/utils/threads/thread.h

```diff
@@ -220,6 +235,34 @@ class Thread {
   sync_primitives::ConditionalVariable state_cond_;
 
  private:
+  void SetState(const ThreadState thread_state) {
+    thread_state_ = thread_state;
+  }
+
+  void SwitchOnState(const ThreadState thread_state) {
+    if (IsSwitchOnState(thread_state)) {
+      return;
+    }
+
+    thread_state_ ^= thread_state;
+  }
+
+  void SwitchOffState(const ThreadState thread_state) {
+    if (!IsSwitchOnState(thread_state)) {
+      return;
+    }
+
+    thread_state_ ^= thread_state;
+  }
+
+  bool IsSwitchOnState(const ThreadState thread_state) const {
```

Comment:
Add description. It is not obvious what this funciton should return

Reply:
@LuxoftAKutsan 
Fixed in: https://github.com/LuxoftSDL/sdl_core/pull/50/commits/8cf511b7422bec98ecb72f4a7fc30c46e62c8896